### DISCUSSION TITLE
Rev 23 boot tape not working?

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#13] [General]
As a distraction from reading Subroutines Ref guides, I thought I'd start a clean install of Rev 23 just for the hell of it. 

Rev23 from https://sysovl.info/downloads_prime_primedist.html 
Prime Emulator ver ebf5511 May  5 2020
Following https://sysovl.info/reference_prime_drb_installing_primos233.html
Hosted on Pi4 Raspbian GNU/Linux 10 (buster)

But here's what I get when I try to boot ...

```
pi@sandsend:~/PRIME/PIP02 $ ls -l
total 93724
-rwxr--r-- 1 pi pi     5422 Jan 29 11:16 cfg233.tap
-rwxr--r-- 1 pi pi   189112 May  5  2020 em
-rwxr--r-- 1 pi pi  6514310 Apr 10  2020 m234bt.tap
-rwxr--r-- 1 pi pi  1412338 Apr  8  2020 m234c1.tap
-rwxr--r-- 1 pi pi 18220270 Apr  8  2020 m234ic.tap
-rwxr--r-- 1 pi pi 37607036 Apr  8  2020 m234u1.tap
-rwxr--r-- 1 pi pi  7942380 Apr  8  2020 m234u2.tap
-rwxr--r-- 1 pi pi  2143802 Apr  8  2020 m234v1.tap
lrwxrwxrwx 1 pi pi       10 Jan 29 11:45 mt0 -> m234bt.tap
-rwxr--r-- 1 pi pi 21916851 Jan 29 11:11 primos_23.4.tar.gz
pi@sandsend:~/PRIME/PIP02 $ ./em -boot 10005
[Prime Emulator ver ebf5511 May  5 2020]
[Copyright (C) 2005-2019 Jim Wilcoxson prirun@gmail.com]
Booting from file mt0

Fatal error: CPU halt
instruction #92 at 0/0  ^000000^
A='0/0  B='0/0  L='0/0  X='177730/-40 K=14000 [64VF]
owner=0 , modals=0 [I0N]
Check error.log for more information

pi@sandsend:~/PRIME/PIP02 $ cat error.log
-nport is zero, PNC not started
emulator: device '7 failed initialization - device removed
-tport is zero, AMLC devices not started
emulator: device '53 failed initialization - device removed
emulator: device '54 failed initialization - device removed
pi@sandsend:~/PRIME/PIP02 $

```

The tape reads OK on my patch-work R20 system, and contains (via regular MAGRST) ...

```
 tree -nw
[TREE rev 1.2-22.1 Copyright (c) Prime Netherlands, 1990]

*>MAGRST.RUN (dam)
*>FIX_DISK.RUN (dam)
*>MAKE.SAVE (sam)
*>T$0000 (sam)
*>PRIRUN (ufd)
        >PRIMOS.SAVE (sam)
        >VERSION_STAMP.CPL (sam)
        >PR0000 (sam)
        >PR0004 (sam)
        >PR0006 (sam)
        >PR0007 (sam)
        >PR0010 (sam)
        >PR0011 (sam)
        >PR0012 (sam)
        >PR0013 (sam)
        >PR0014 (sam)
        >PR0015 (sam)
        >PR0016 (sam)
        >PR0017 (sam)
        >PR0020 (sam)
        >PR0021 (sam)
        >PR0031 (sam)
        >PR0032 (sam)
        >PR0033 (sam)
        >PR0034 (sam)
        >PR0037 (sam)
        >PR0040 (sam)
        >PR0041 (sam)
        >PR0044 (sam)
        >PR0045 (sam)
        >PR0046 (sam)
        >PR0050 (sam)
        >PR0051 (sam)
        >PR0055 (sam)
        >PR0056 (sam)
        >PR0064 (sam)
        >PR0065 (sam)
        >PR0067 (sam)
        >PR0070 (sam)
        >PR0141 (sam)
        >PR0142 (sam)
        >PR0036 (sam)
        >PR6000 (sam)
        >PR6003 (sam)
        >PR0052 (sam)
        >PR0207 (sam)
        >PR0047 (sam)
        >PR0053 (sam)
        >PR0675 (sam)
        >PR0300 (sam)
        >PRIMOS.COMI.TEMPLATE (sam)
        >PR0600 (sam)
        >PR0477 (sam)
        >PR0500 (sam)
        >PR0203 (sam)
        >PR0204 (sam)
        >PR0063 (sam)
        >RING0.MAP (sam)
        >RING3.MAP (sam)
        >PR0250 (sam)
        >PR0623 (sam)
        >PR0251 (sam)
        >PR0066 (sam)
        >PR0240 (sam)
        >PR0104 (sam)


```

Is the tape unbootable, or have I done something dumb?

**Comment by drboone** [comment]
I went several rounds trying to get the boot tapes right, so its entirely possible there's something wrong with that one.  I'll do a test later and see if I need to go a few more rounds. :)

**Comment by SirNibble** [comment]
Tried the Rev 23.2 from https://yagi.h-net.org/ and get ..

```
Booting from file mt0

Fatal error: Program start > ending: boot image is corrupt
instruction #0 at 0/0 SEG0 ^000000^
A='0/0  B='0/0  L='0/0  X='0/0 K=0 [16SF000/0 ]
owner=0 SEG0, modals=0 [I0N]
Check error.log for more information

```

I'm impressed you've done as many rounds as you have!

**Comment by drboone** [comment]
The ones at sysovl.info are slightly newer, so maybe my mistake was not keeping multiple web points in sync.

**Comment by drboone** [comment]
Confirmed, both of the 23.x tapes at sysovl.info are busted, and probably even more of the stuff on yagi.h-net.org.  I'll get that fixed.

The 21.x and 24.x tapes from sysovl.info seem to be more bootable.

**Comment by drboone** [comment]
As I was digging around working on the boot record problem, I rediscovered a note I left myself about how the 23.2 set is somewhat broken, which is why it isn't listed on sysovl.info.  I did update the boot record, and it is in the directory on yagi, but I'm not sure it's worth spending time on.

I have uploaded the 23.3 set.  I was able to boot from this, make the pack, restore it all, and reboot from disk, so I think it's in decent shape.  I'll work on the 23.4 set next.

**Comment by SirNibble** [comment]
Perfect!

Have an issue with erase, though.

Using PUTTY as my client, and set CONFIG - ERASE to 377 for PUTTY's defaul xterm PC backspace. 

Problem is is that the cursor doesn't actually move back and remove a character. R19 -> 22 work as expected, but not R23

The default 'CTRL-H' (ERASE 210) moves the cursor back, but doesn't remove a character.

Do I need another CONFIG change?

Also, you may want to note in https://sysovl.info/reference_prime_drb_installing_primos233.html that the cfg233,tap over-writes the System Name unless you remember to edit CONFIG - SYSNAM

**Reply by drboone** [reply]
BTW If you boot without CONFIG, PRIMOS prompts for a few key settings but only stores those in RAM.

**Comment by drboone** [comment]
Base PRIMOS doesn't do cleanup as you might see in unix where it echoes backspace-space-backspace.  If ECL is running, it will do the cleanup.  Are you seeing ECL not do it?

**Comment by SirNibble** [comment]
ECL works as expected with both CTRL-H and Backspace. So that's nice!

Ta!

**Comment by SirNibble** [comment]
Still getting some strangeness with 'erase' - these are all via a PuTTY SSH session to a Pi4.

Console command prompt with ECL works as expected, but in commands (eg MAGRST or ED etc) 'erase 377 /* backspace' doesn't move the cursor back, but 'erase 210 /* ctrl-h'  does. (ditto for R22 & R24)

R19 -> R21 don't show the same behaviour. 

Telnet into the emulated servers as a user at any revision with either erase config works as expected. (No change in PuTTY settings apart from Telnet rather than SSH)

Can work round it, but it is odd.

**Comment by drboone** [comment]
Oh, right.  Subsystems don't necessarily use ECL for input.  I know DBG at rev23 does, for example, but I don't think MAGSAV/MAGRST ever did.

Assuming ECL is not involved, and echo is on, PRIMOS is going to echo back whatever character you type, and the terminal will react to that.  Pretty much no terminal does anything visibly useful in response to a DEL character.  A BS does what you would expect on most.

Most of the time I spent on Primes, they were working hard enough that character echo was a bit slow, and I was therefore basically forced to learn to count keystrokes and visualize the results in my head anyway.  We also used non-vt100-family terminals, most of which use BS as the backspace character.  (For us, tvi925 emulation.)

**Comment by SirNibble** [comment]
I've put it back to the default 210 and set-up slighty different PuTTY profiles, so it's all peachy now. (Complete with Green or Amber depending on which server I'm talking to.)

Mulling over this, it rings faint memory bells of when we started introducing PCs at work and getting ProComm to act how we had our Pericomm terminals behaving.

**Comment by drboone** [comment]
If your users were trained to use the backspace key on the Pericom terminals, then switching to PC-based emulation which often wants to send DEL for the <----- key, would definitely have required an adjustment.

I don't think we got many Pericom terminals on this side of the pond.  I certainly wasn't familiar with them.

**Comment by SirNibble** [comment]
Most of the original Pericomm DEC VT100 clones I managed were a  PITA to support as crappy DIP switch packages had to be reset and contact-cleaner sprayed for them to 'remember' their settings. Sneeze near the blasted things and they'd change. Or the DB25 would fall off the board. 

Handy settings cheat-sheet under the keyboard, which was handy.

The later amber ones were bullet proof, though. I'd love to get one, but suspect one's signifcant other may consider the £££ not as much as an upgrade as me! ;)
